lua/packer/plugin_utils.lua

```diff
@@ -96,26 +96,44 @@ plugin_utils.ensure_dirs = function()
 end
 
 plugin_utils.find_missing_plugins = function(plugins, opt_plugins, start_plugins)
-  if opt_plugins == nil or start_plugins == nil then
-    opt_plugins, start_plugins = plugin_utils.list_installed_plugins()
-  end
+  return a.sync(function ()
+    if opt_plugins == nil or start_plugins == nil then
+      opt_plugins, start_plugins = plugin_utils.list_installed_plugins()
+    end
 
-  -- NOTE/TODO: In the case of a plugin gaining/losing an alias, this will force a clean and
-  -- reinstall
-  local missing_plugins = {}
-  for _, plugin_name in ipairs(vim.tbl_keys(plugins)) do
-    local plugin = plugins[plugin_name]
+    -- NOTE/TODO: In the case of a plugin gaining/losing an alias, this will force a clean and
+    -- reinstall
+    local missing_plugins = {}
+    for _, plugin_name in ipairs(vim.tbl_keys(plugins)) do
+      local plugin = plugins[plugin_name]
 
-    local plugin_path = util.join_paths(config[plugin.opt and 'opt_dir' or 'start_dir'],
-                                        plugin.short_name)
-    local plugin_installed = (plugin.opt and opt_plugins or start_plugins)[plugin_path]
+      local plugin_path = util.join_paths(config[plugin.opt and 'opt_dir' or 'start_dir'],
+      plugin.short_name)
+      local plugin_installed = (plugin.opt and opt_plugins or start_plugins)[plugin_path]
 
-    if not plugin_installed or plugin.type ~= plugin_utils.guess_dir_type(plugin_path) then
-      table.insert(missing_plugins, plugin_name)
+      await(vim.schedule)
+      local guessed_type = plugin_utils.guess_dir_type(plugin_path)
+      if not plugin_installed or plugin.type ~= guessed_type then
+        table.insert(missing_plugins, plugin_name)
+      end
+      if guessed_type == plugin_utils.git_plugin_type then
+        local r = await(plugin.remote_url())
+        local remote = r.ok.remote
+        if r.ok and r.ok.remote then
+          local parts = vim.split(remote, "/")
+          local repo_name = util.join_paths(unpack(vim.list_slice(parts, #parts - 1, #parts)))
+          repo_name = repo_name:gsub("%.git", "")
+          if repo_name and repo_name ~= plugin.name then
+            table.insert(missing_plugins, plugin_name)
+          end
+        end
+      end
     end
-  end
 
-  return missing_plugins
+    -- TODO why do we need to await vim.schedule here
+    await(vim.schedule)
```

Comment:
@wbthomason without this line subsequent functions which use `vim.fn` or `nvim_exec` all fail. I'm not sure exactly what's going on but I found that putting this here fixed those issues 😓

Reply:
Try `await(main)` - it's the same thing, but semantically a bit clearer. This is necessary to ensure that you're on the main loop, i.e. to do things outside of `api-fast`.